src/main/groovy/nebula/plugin/publishing/PublicationBase.groovy

```diff
@@ -0,0 +1,87 @@
+package nebula.plugin.publishing
+
+import nebula.plugin.publishing.ivy.IvyBasePublishPlugin
+import nebula.plugin.publishing.maven.MavenBasePublishPlugin
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.publish.ivy.IvyPublication
+import org.gradle.api.publish.ivy.plugins.IvyPublishPlugin
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin
+
+trait PublicationBase implements Plugin<Project>, StandardTextValues {
+
+    /**
+     * Checks to see if the task already exists or not on the project.  If it does not, then it creates it.
+     * @param project project to add the task too
+     * @param task Map configuration of the task to create
+     * @return Null if it was not created, the task object if it was.
+     */
+    Task addTaskLocal(Project project, Map<String, ?> task) {
+        if (!project.tasks.findByName(task.get('name').toString())) {
+            project.tasks.create(task)
+        } else {
+            null
+        }
+    }
+
+    void checkAddMavenPublish(Project project, Task createdTask) {
+        project.plugins.withType(MavenPublishPlugin) {
+            project.plugins.apply(MavenBasePublishPlugin)
+
+            project.publishing.publications {
```

Comment:
https://github.com/researchgate/gradle-release/issues/125

https://stackoverflow.com/questions/28020520/custom-gradle-plugin-causes-cannot-configure-the-publishing-extension

are where i got the fix for the "can not modify publish after its been accessed" error.  As i said, its a well known gradle/groovy issue in their maven-publish plugin.  

Still trying to figure out a way to boil it down to a unit test.  the conditions are pretty specific and not easy to replicate.